circuit-macros/src/circuit_type/secret_share_types.rs

```diff
@@ -141,58 +127,15 @@ fn build_addition_impl(base_type: &ItemStruct) -> TokenStream2 {
 
     let output_type_name = ident_with_generics(base_type.ident.clone(), generics.clone());
 
-    // Implementation calls out to
+    // Implementation calls out to the `add_shares`
     let impl_block: ItemImpl = parse_quote! {
         impl #generics Add<#secret_share_type_name> for #secret_share_type_name
             #where_clause
         {
             type Output = #output_type_name;
 
             fn add(self, rhs: #secret_share_type_name) -> Self::Output {
-                self.add_shares(rhs)
-            }
-        }
-    };
-    impl_block.to_token_stream()
-}
-
-/// Build an addition implementation on secret share variable types
-fn build_var_type_addition_impl(secret_share_type: &ItemStruct) -> TokenStream2 {
```

Comment:
Out of curiosity, why do we remove this `Add` impl? We still have variable types for secret-shared objects allocated in the proof system, so this could still be used, right?

Reply:
We need access to the circuit to add variables now (dalek just generated a linear combination). We can't pass the circuit into the Add interface, and it's a bit too complicated to make all variables hold references to the circuit, so we just remove the Add impl and use add_shares instead